knowledge-base/how-to-use-kendodatetimepicker-for-parameter-editors-in-the-angular-report-viewer.md

```diff
@@ -0,0 +1,120 @@
+---
+title: How to Use KendoDateTimePicker for Parameter Editors in the Angular Report Viewer
+description: See how to use the KendoDateTimePicker for selecting date and time for the report parameter by creating custom parameter editor in the Angular Report Viewer
+type: how-to
+page_title: Using KendoDateTimePicker in Angular Report Viewer
+slug: how-to-use-kendodatetimepicker-for-parameter-editors-in-the-angular-report-viewer
+position: 
+tags: angular, reportviewer, kendodatetimepicker
+ticketid: 1543447
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+		<tr>
+			<td>Project Type</td>
+			<td>Angular</td>
+		</tr>
+		<tr>
+			<td>Report Viewer</td>
+			<td>Angular Report Viewer</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+This article describes how to use the [KendoDateTimePicker](https://docs.telerik.com/kendo-ui/api/javascript/ui/datetimepicker) as a custom parameter editor
+in [Angular Report Viewer](../angular-report-viewer). This approach allows the selection of date and time for the **DateTime** [report parameters](../designing-reports-parameters).
+
+
+## Solution
+> Telerik kendoDateTimePicker widget is not included in the Kendo UI JavaScript distributed by 
+Telerik Reporting through telerikReportViewer.kendo-x.x.x.min.js file, or kendo.subset.2015.3.930.min.js in older versions.
+For that reason, you will have to replace the Kendo UI subset with the full Kendo UI JavaScript e.g. kendo.all.min.js.
+
+Also, you need to make sure that Kendo all is loaded after jQuery. Because of that, we will use a special logic that sets the visibility
+of the viewer after Kendo all is loaded:
+
+1. In the initializaion of the viewer, specify the parameterEditors [option](../angular-report-viewer-api-options).
+```TypeScript
+<tr-viewer #viewer1 *ngIf="visible"
+    [containerStyle]="viewerContainerStyle"
+    [serviceUrl]="'http://localhost:59654/api/reports/'"
+    [reportSource]="{
+        report: 'MyReport.trdp',
+        parameters: {}
+    }"
+    [parameterEditors]="[{
+        match: match,
+        createEditor: createEditor
+    }]"
+</tr-viewer>
+```
+
+2.Then in the **app.component.ts** implement the createEditor function. You see that we set the visiblity of the viewer
+to True once kendo.all.min.js is loadded. This is required because firstly jQuery has to be loaded and then kendo.all.min.js:
```

Comment:
The text "You see that we set the visiblity of the viewer to True once kendo.all.min.js is loadded. This is required because firstly jQuery has to be loaded and then kendo.all.min.js:" repeats the same that was said above. There are also typos in it.